examples/lp503x/lp503x_main.c

```diff
@@ -0,0 +1,714 @@
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "system/readline.h"
+
+#include <nuttx/leds/lp503x.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define LP503X_HELP_TEXT(x) x
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+typedef int (*lp503x_func)(char *pargs);
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int lp503x_cmd_quit(char *parg);
+static int lp503x_cmd_help(char *parg);
+static int lp503x_cmd_set_rgbled(char *parg);
+static int lp503x_cmd_setled(char *parg);
+static int lp503x_cmd_setcolour(char *parg);
+static int lp503x_cmd_pattern(char *parg);
+static int lp503x_cmd_brightness(char *parg);
+static int lp503x_cmd_mode(char *parg);
+static int lp503x_cmd_disable(char *parg);
+static int lp503x_cmd_set_bank_mode(char *parg);
+static int lp503x_cmd_set_banka_colour(char *parg);
+static int lp503x_cmd_set_bankb_colour(char *parg);
+static int lp503x_cmd_set_bankc_colour(char *parg);
+static int lp503x_cmd_set_bank_brightness(char *parg);
+static int lp503x_cmd_set_individual_mode(char *parg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+int fd;
+struct lp503x_cmd_s
+{
+  const char *cmd;       /* The command text           */
+  const char *arghelp;   /* Text describing the args   */
+  lp503x_func  pfunc;    /* Pointer to command handler */
+  const char *help;      /* The help text              */
+};
```

Comment:
let's move this to `Private Types` section

Reply:
OK